### PR TITLE
Middleware

### PR DESCRIPTION
**Pull request by ReubenKabiti**
Added the ability to add middleware.

usage example

```js
import merlee from "merlee.js"

const mw1 = function (req, res, next) { 
    console.log("m1!");
    req.name = "name set from req!";
    next();
}

const mw2 = function (req, res, next) { 
    console.log("m2!");
    res.name = req.name;
    next();
}

const app = merlee({
    port: 3000,
    middleware: [mw1, mw2]
});

app.listen((p) => console.log("listening on port: " + p));

app.handler({ path: "/",  method: "GET" }, (req, res) => {
    console.log("name: " + res.name);
    res.send("hello world");
});
```

### CODE REVIEW

File: lib/merlee.js [hunk 1]
```diff
@@ -61,6 +56,32 @@ class Merlee {
...
+        if (cmw < midw.length - 1) {
+          cmw++;
+          midw[cmw](req, res, next);
+        } else if (cmw == midw.length - 1) {
+          calledNextAtEnd = true;
+        }
+      }
+
+      if (midw) {
+        if (midw.length > 0) {
```
Comment (mwenku) [inline lib/merlee.js]:
```suggestion
        if (midw && midw?.length > 0) {
```
if know you know what i mean. the `?` should only run when the class passed has a length method'ish i know you know what i mean though lol

Comment (ReubenKabiti) [reply lib/merlee.js]: Oh yeah makes sense

File: lib/merlee.js [hunk 2]
```diff
@@ -61,6 +56,32 @@ class Merlee {
...
+      const next = () => {
+        if (cmw < midw.length - 1) {
+          cmw++;
+          midw[cmw](req, res, next);
+        } else if (cmw == midw.length - 1) {
+          calledNextAtEnd = true;
+        }
+      }
+
+      if (midw) {
```
Comment (mwenku) [inline lib/merlee.js]:
```suggestion

```

File: lib/merlee.js [hunk 3]
```diff
@@ -61,6 +56,32 @@ class Merlee {
...
+        if (midw.length > 0) {
+          midw[0](req, res, next);
+          if (!calledNextAtEnd) {
+            return; // return if next was not called at the end of the middleware chain
+          }
+        }
+      }
```
Comment (mwenku) [inline lib/merlee.js]:
```suggestion
      
```

**Review by mwenku — CHANGES_REQUESTED** [review]
Can you check this part of the code. you don't need two condition, 

one should be enough

**Review by mwenku — APPROVED** [review]
Looks good to me 👍

**Pull request merged**